internal/resources/webform/webform.js

```diff
@@ -2181,10 +2222,15 @@ window.initGRPCForm = function(services, invokeURI, metadataURI, debug) {
                 t.tabs("enable", 2);
                 t.tabs("option", "active", 2);
             })
-            .fail(function(data, status) {
+            .fail(function(failureData, status) {
+                addHistory({
+                    ...history,
+                    duration: window.performance.now() - startTime,
+                    result: `failure: ${status}`
```

Comment:
Why not instead store `failureData` and `status`, instead of stringifying? What types will they be? More structure means we could make interesting improvements later w/out worrying about local storage having older representation and having to support multiple versions of data.